### PR TITLE
Adds tab function to docs and typings

### PR DESCRIPTION
**Pull request by GentlemanHal**
Just a simple patch to add the tab function to docs and typings.

### CODE REVIEW

File: typings/index.d.ts [hunk 1]
```diff
@@ -15,6 +15,7 @@ declare const userEvent: {
         text: string,
         userOpts?: IUserOptions
     ) => Promise<void>;
+    tab: (userOpts?: { shift: boolean }) => void;
```
Comment (YuanchengWu) [inline typings/index.d.ts]: This type is no longer up to date. Could you add `focusTrap` to it?
better yet, define an interface and add defaults to the options

```ts
interface TabUserOptions {
  shift: boolean = false;
  focusTrap: Document = document;
}

declare const userEvent: {
  ...
  tab: (userOpts?: TabUserOptions) => void;
};

Comment (GentlemanHal) [reply typings/index.d.ts]: Should the type of `focusTrap` be `TargetElement` which is defined as `type TargetElement = Element | Window;`?

Comment (GentlemanHal) [reply typings/index.d.ts]: Or maybe it needs to be `Document | Element`?

Comment (GentlemanHal) [reply typings/index.d.ts]: Looking at `lib.dom.d.ts` `querySelectorAll` is defined in `ParentNode` which `Document` and `Element` both extend, so `Document | Element` seems like the correct choice.

Comment (Gpx) [reply typings/index.d.ts]: Are you happy with the current state?

Comment (GentlemanHal) [reply typings/index.d.ts]: I am 👍

Comment (Gpx) [reply typings/index.d.ts]: Cool, then I'm going to merge it

**Pull request merged**